scality_manila_utils/cli.py

```diff
@@ -98,109 +99,163 @@ def main(args=None):
-        'export_name',
-        help='Filesystem to grant access to'
-    )
-    parser_grant.add_argument(
-        'host',
-        help='IP address or network to grant access for'
-    )
-    parser_grant.add_argument(
-        'options',
-        help='Export options',
-        nargs='*'
-    )
-
-    parser_revoke = subparsers.add_parser(
-        'revoke',
-        help='Revoke access from an existing filesystem, and reexport it'
-    )
-    parser_revoke.add_argument(
-        'export_name',
-        help='Filesystem to revoke access from'
-    )
-    parser_revoke.add_argument(
-        'host',
-        help='IP address or network to revoke access for'
-    )
-
-    parser_check = subparsers.add_parser(
-        'check',
-        help='Check for required binaries and running services'
-    )
-
-    parser_get = subparsers.add_parser(
-        'get',
-        help='List the addresses that a filesystem is exported to'
-    )
-    parser_get.add_argument(
-        'export_name',
-        help='Filesystem to get information about'
-    )
-
-    parser_wipe = subparsers.add_parser(
-        'wipe',
-        help='Remove an existing export. The export must not have any grants.'
+    nfs_parser.add_argument(
+        '--exports-file',
+        help='Path to NFS exports file',
+        default='/etc/exports.conf',
     )
-    parser_wipe.add_argument(
-        'export_name',
-        help='Filesystem to remove'
+    nfs_parser.add_argument(
+        '--root-export',
+        help='NFS export path to the ring NFS root volume',
+        default='127.0.0.1:/'
     )
-
-    parser_create.set_defaults(func=scality_manila_utils.helper.add_export)
-    parser_grant.set_defaults(func=scality_manila_utils.helper.grant_access)
-    parser_revoke.set_defaults(func=scality_manila_utils.helper.revoke_access)
-    parser_get.set_defaults(func=scality_manila_utils.helper.get_export)
-    parser_wipe.set_defaults(func=scality_manila_utils.helper.wipe_export)
-    parser_check.set_defaults(
-        func=scality_manila_utils.helper.verify_environment
+    smb_parser.add_argument(
+        '--root-export',
+        help='SOFS directory that holds the SMB shares',
+        default='/ring/fs'
     )
 
-    parsed_args = command_parser.parse_args(args)
+    nfs_subparsers = nfs_parser.add_subparsers()
+    smb_subparsers = smb_parser.add_subparsers()
+
+    protocol_subparsers = [('nfs', nfs_subparsers), ('smb', smb_subparsers)]
+
+    def add_parser_create():
+        for (protocol, subparsers) in protocol_subparsers:
+            help = description = 'Prepare an export without any access grants'
+            parser_create = subparsers.add_parser(
+                'create', description=description,
+                help=help
+            )
+            parser_create.add_argument(
+                'export_name',
+                help='Export point to create'
+            )
+            parser_create.set_defaults(protocol=protocol, func='add_export')
```

Comment:
To ensure all implementation modules (`nfs_helper`, `smb_helper`) actually implement all these functions, without relying on runtime-testing, but instead causing a failure immediately, I guess you could do something like

```python
protocol_subparsers = [ ('nfs', nfs_subparsers, scality_manila_utils.nfs_helpers)
                      , ('smb', smb_subparsers, scality_manila_utils.smb_helpers)
                      ]

def add_parser_create():
    for (protocol, subparsers, helper) in protocol_subparsers:
        ...
        parser_create.set_defaults(protocol=protocol, func=getattr(helper, 'add_export'))
```
Maybe `protocol` is no longer required then (didn't read all the way through the PR yet).